default/AI/MilitaryAI.py

```diff
@@ -16,6 +17,34 @@
 totMilRating = 0
 num_milships = 0
 verbose_mil_reporting = False
+best_ship_rating_cache = {}  # indexed by turn, value is rating of that turn
+
+
+def cur_best_mil_ship_rating(include_designs=False):
+    """Find the best military ship we have available in this turn and return its rating.
```

Comment:
I expect he has that option in there because the current functionality checks for the rating of the best buildable design, not the rating of the best existing ship, and he wants all of us to be able to test it both ways.  I am inclined to think that the default should be to include designs, and at the very least the call to this via calculateMilitaryPriority() should use the form that includes designs, because it is trying to make a calculation about building new ships.  Some of the other uses I could kind of see going either way, although I think my preference is still to go with best buildable design rating like it was.   Please point out and explain any particular spots where you think it better to use the rating of the best existing ship.